src/main/scala/wdl4s/Task.scala

```diff
@@ -30,7 +31,87 @@ object Task {
       case x: Ast => ParameterCommandPart(x, wdlSyntaxErrorFormatter)
     }
 
-    val outputs = ast.findAsts(AstNodeName.Output) map { TaskOutput(_, declarations, wdlSyntaxErrorFormatter) }
+    val variablesReferencedInCommand = for {
+      param <- commandTemplate collect { case x: ParameterCommandPart => x }
+      variable <- param.expression.variableReferences
+    } yield variable
+
+    variablesReferencedInCommand foreach { variable =>
+      if (!declarations.map(_.name).contains(variable.getSourceString)) {
+        throw new SyntaxError(wdlSyntaxErrorFormatter.commandExpressionContainsInvalidVariableReference(taskNameTerminal, variable))
+      }
+    }
```

Comment:
I think this would be better as `Validation`s that accumulate errors rather than failing fast on the first problem